enterprise/frontend/src/metabase-enterprise/metabot/components/MetabotAdmin/MetabotPurchasePage/MetabotPurchasePage.tsx

```diff
@@ -0,0 +1,184 @@
+import { useDisclosure } from "@mantine/hooks";
+import { type ReactElement, useCallback } from "react";
+import { t } from "ttag";
+import * as Yup from "yup";
+
+import {
+  SettingsPageWrapper,
+  SettingsSection,
+} from "metabase/admin/components/SettingsSection";
+import { getCurrentUser } from "metabase/admin/datamodel/selectors";
+import {
+  useListAddOnsQuery,
+  usePurchaseCloudAddOnMutation,
+} from "metabase/api";
+import ExternalLink from "metabase/common/components/ExternalLink";
+import { useSetting } from "metabase/common/hooks";
+import {
+  Form,
+  FormCheckbox,
+  FormProvider,
+  FormSubmitButton,
+} from "metabase/forms";
+import { useSelector } from "metabase/lib/redux";
+import { Alert, Card, Icon, Loader, Stack, Text } from "metabase/ui";
+import { useGetBillingInfoQuery } from "metabase-enterprise/api";
+
+import { MetabotSettingUpModal } from "../MetabotSettingUpModal";
+import { handleFieldError, isFetchBaseQueryError } from "../utils";
+
+import { MetabotRadios } from "./MetabotPurchasePage.components";
+import type { MetabotPurchaseFormFields } from "./types";
+
+const validationSchema = Yup.object({
+  quantity: Yup.string(),
+  terms_of_service: Yup.boolean(),
+});
+
+export function MetabotPurchasePage(): ReactElement {
+  const currentUser = useSelector(getCurrentUser);
+  const tokenStatus = useSetting("token-status");
+  const storeUserEmails =
+    tokenStatus?.["store-users"]?.map(({ email }) => email.toLowerCase()) ?? [];
+  const isStoreUser = storeUserEmails.includes(
+    currentUser?.email.toLowerCase(),
+  );
+  const anyStoreUserEmailAddress =
+    storeUserEmails.length > 0 ? storeUserEmails[0] : undefined;
```

Comment:
!low priority - feel free to skip

this seems nice to put in hook so we could reuse in the future.